### ISSUE TITLE
GitHub homepage paragraph text wrong colour

### ISSUE THREAD

**Issue by atouchet** [servo/servo#17464]
The text is supposed to show up grey but in Servo it is white.

Servo:

![Servo text](https://i.imgur.com/F81D8dV.png)

Firefox:

![Firefox text](https://i.imgur.com/saDNKkx.png)

Tested in Windows 10.

**Comment by jdm** [comment]
Specifically, the CSS contains `color: rgb(255, 255, 255, 0.6)`, so we're apparently not treating the alpha correctly.

**Comment by jdm** [comment]
```html
<style>body { background-color: black }</style>
<p style="color: rgb(255, 255, 255, 0.6)">Github is a development platform<p>
<p style="color: rgb(255, 255, 255)">Github is a development platform</p>
```
Firefox:
<img width="241" alt="screen shot 2017-06-21 at 11 32 47 pm" src="https://user-images.githubusercontent.com/27658/27416213-f4d948fc-56d9-11e7-898a-e2bdee311e93.png">
Servo:
<img width="227" alt="screen shot 2017-06-21 at 11 33 05 pm" src="https://user-images.githubusercontent.com/27658/27416222-fef07cd4-56d9-11e7-9b5d-2573516c58c3.png">

**Comment by emilio** [comment]
I've verified that the text arrives with the correct color to the display, list.

Furthermore, I've verified that the color is still correct in WR's TextRunPrimitiveCpu::write_gpu_blocks.

And, if I apply the following diff in WR, which i'd expect to show a bunch of squares with the text color and the correct alpha:

```
diff --git a/webrender/res/ps_text_run.fs.glsl b/webrender/res/ps_text_run.fs.glsl
index b21ac22..40a41ff 100644
--- a/webrender/res/ps_text_run.fs.glsl
+++ b/webrender/res/ps_text_run.fs.glsl
@@ -18,4 +18,5 @@ void main(void) {
     alpha = min(alpha, do_clip());
     oFragColor = vec4(vColor.rgb, vColor.a * alpha);
 #endif
+    oFragColor = vec4(vColor.rgb, vColor.a);
 }
```

I indeed see the squares, but I still see no alpha.

@glennw, this seems to be a WR bug, and it seems to me it'd be much easier to you to diagnose than to me... Maybe a premultiplication issue? That'd be weird...

**Comment by glennw** [comment]
My guess (unconfirmed) is that it's related to having alpha on sub-pixel enabled text. We need to do a bit more work on sub-pixel text support to handle this case.

**Comment by stshine** [comment]
I think we should close this after the webrender fix landed in servo.

**Comment by atouchet** [comment]
This is fixed in Servo as of #17694. Thanks!
